### PR TITLE
[AMORO-2897] Mask the sensitive info for the Terminal output logs

### LINKED ISSUE #2897: [Improvement]: Mask the sensitive info for the Terminal output logs
### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/amoro/issues?q=is%3Aissue) and found no similar issues.

### What would you like to be improved?

Logs output contain some sensitive information such as s3 secret key, which should be masked.

image

### How should we improve?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Subtasks

_No response_

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by tcodehuber**
## Why are the changes needed?

Close #2897.

## Brief change log

-

## How was this patch tested?

- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] Run test locally before making a pull request

## Documentation

- Does this pull request introduce a new feature? (yes / no)
- If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

File: amoro-ams/amoro-ams-server/src/main/java/org/apache/amoro/server/dashboard/utils/DesensitizationUtil.java [hunk 1]
```diff
@@ -22,11 +22,26 @@
...
+
   public static String desensitize(Object data) {
     return COMMON_SECRET_FORMAT;
   }
 
   public static boolean isDesensitized(Object data) {
     return COMMON_SECRET_FORMAT.equals(data);
   }
+
+  public static boolean containsSensitiveVal(String input) {
```
Comment (xleoken) [inline amoro-ams/amoro-ams-server/src/main/java/org/apache/amoro/server/dashboard/utils/DesensitizationUtil.java:35]: We can define `SENSITIVE_CONF_KEYWORDS` as a list, then use contains method.

File: amoro-ams/amoro-ams-server/src/main/java/org/apache/amoro/server/dashboard/utils/DesensitizationUtil.java [hunk 2]
```diff
@@ -22,11 +22,26 @@
 public class DesensitizationUtil {
   public static String COMMON_SECRET_FORMAT = "********";
 
+  private static final String[] SENSITIVE_CONF_KEYWORDS = {"secret", "token", "password"};
```
Comment (zhoujinsong) [inline amoro-ams/amoro-ams-server/src/main/java/org/apache/amoro/server/dashboard/utils/DesensitizationUtil.java:25]: I am not sure if we should desensitize all properties containing the keywords above.
Or we may better define specific properties we should desensitize on?

Comment (tcodehuber) [reply amoro-ams/amoro-ams-server/src/main/java/org/apache/amoro/server/dashboard/utils/DesensitizationUtil.java:25]: Good idea, i will try to do that.

File: amoro-ams/amoro-ams-server/src/main/java/org/apache/amoro/server/terminal/TerminalSessionContext.java [hunk 3]
```diff
@@ -48,6 +52,8 @@ public class TerminalSessionContext {
   private volatile ExecutionTask task = null;
   private final TerminalSessionFactory factory;
   private final Configurations sessionConfiguration;
+
+  private final List sensitiveConfKeys;
```
Comment (zhoujinsong) [inline amoro-ams/amoro-ams-server/src/main/java/org/apache/amoro/server/terminal/TerminalSessionContext.java:56]:
```suggestion
  private final List<String> sensitiveConfKeys;
```

File: amoro-ams/amoro-ams-server/src/main/java/org/apache/amoro/server/terminal/local/LocalSessionFactory.java [hunk 4]
```diff
@@ -50,10 +54,16 @@ public class LocalSessionFactory implements TerminalSessionFactory {
 
   SparkSession context = null;
   Configurations conf;
+  List sensitiveConfKeys;
```
Comment (zhoujinsong) [inline amoro-ams/amoro-ams-server/src/main/java/org/apache/amoro/server/terminal/local/LocalSessionFactory.java:57]:
```suggestion
  List<String> sensitiveConfKeys;
```

**Review by zhoujinsong — COMMENTED** [review]
@tcodehuber I left some small suggestions. PTAL.

**Comment by tcodehuber** [comment]
> @tcodehuber I left some small suggestions. PTAL.

Done.

**Review by zhoujinsong — APPROVED** [review]
LGTM. Thanks for the contribution!

**Pull request merged**